test/service.go

```diff
@@ -65,9 +65,41 @@ func CreateService(ctx *Context, name, namespace, image string) (*servingv1beta1
 	return service, nil
 }
 
+func GetPod(ctx *Context, ns string) {
+	// run infinite for loop because once after the Global proxy update operator pod will be recreated
+	// and which intern updates knative serving controller pod so we don't no exact time duration for this to happen
+	for {
```

Comment:
I'd recommend using apimachinery's `wait` package here. `wait.PollImmediate` should be a good fit.

Reply:
I can use `wait.PollImmediate` but i am not sure how much time it will take for operator pod to recreate, update env and then operator pod will update those env to Knative **controller** POD and once Knative controller POD is in Running and have Env need to return from that function.

So i  am not able to estimate exact time for this to happen
that's why added infinite for loop

Reply:
Well eventually the tests will just time out then? I think we should set a reasonable upper boundary. Maybe start with 5 minutes? We could in addition log how long it took to dial that value down in the future.

Reply:
I used max 10 min as timeout